### ISSUE TITLE
Toil leader environment pickle-and-restore may be broken and not work on Mesos

### ISSUE THREAD

**Issue by adamnovak** [DataBiosphere/toil#4835]
We're meant to snapshot the environment when the workflow starts up, and then ship that off to the worker to restore when running jobs:

https://github.com/DataBiosphere/toil/blob/0446fe285fe57a3d920d4b5a30d00f2a5aa8589c/src/toil/options/common.py#L680-L682

We capture `os.environ` inside `start()` here, and save to `environment.pickle` in the job store:

https://github.com/DataBiosphere/toil/blob/0446fe285fe57a3d920d4b5a30d00f2a5aa8589c/src/toil/common.py#L898

And we restore it here in the worker:

https://github.com/DataBiosphere/toil/blob/0446fe285fe57a3d920d4b5a30d00f2a5aa8589c/src/toil/worker.py#L192-L228

Now, Slurm's `sbatch` will additionally do its own environment capture and restore, and single-machine Toil also will end up inheriting the leader's environment for free anyway, so it's possible this system is somehow broken and we haven't noticed.

_Originally posted by @adamnovak in https://github.com/DataBiosphere/toil/issues/4773#issuecomment-1991811075_

@glennhickey alleges that this system doesn't actually work and AWS Mesos jobs are not getting the environment variables he sets in his leader script before `toil.start()`.

┆Issue is synchronized with this [Jira Story](https://ucsc-cgl.atlassian.net/browse/TOIL-1520)
┆Issue Number: TOIL-1520

**Comment by adamnovak** [comment]
@glennhickey You said you can reproduce this with:

```
virtualenv --system-site-packes cactus-venv
. cactus-venv/bin/activate
wget https://github.com/ComparativeGenomicsToolkit/cactus/releases/download/v2.7.2/cactus-bin-v2.7.2.tar.gz
tar zxf cactus-bin-v2.7.2.tar.gz
cd cactus-bin-v2.7.2
pip install -U .
cactus aws:us-west-2:glennhickey-jobstore11 ./examples/evolverMammals.txt em.hal --provisioner aws --batchSystem mesos --nodeTypes c4.8xlarge --skipPreprocessor --consCores 4
```

But I'm not able to reproduce the problem on a fixed-sized cluster with Toil 0446fe285fe57a3d920d4b5a30d00f2a5aa8589c. I can start the cluster:

```
TOIL_OWNER_TAG=anovak@soe.ucsc.edu TOIL_APPLIANCE_SELF=quay.io/ucsc_cgl/toil:6.2.0a1-0446fe285fe57a3d920d4b5a30d00f2a5aa8589c-py3.9 toil launch-cluster --keyPairName anovak@swords -T mesos -w 1 --nodeTypes c4.2xlarge --leaderNodeType c4.2xlarge -z us-west-2a adamnovak-test-cluster
```

And then log into it:
```
toil ssh-cluster -z us-west-2a adamnovak-test-cluster
```

And install Cactus like you show, more or less:
```
virtualenv --system-site-packages cactus-venv
. cactus-venv/bin/activate
wget https://github.com/ComparativeGenomicsToolkit/cactus/releases/download/v2.7.2/cactus-bin-v2.7.2.tar.gz
tar zxf cactus-bin-v2.7.2.tar.gz
cd cactus-bin-v2.7.2
git config --global --add safe.directory `pwd`
pip install --upgrade --ignore-installed .
```

And then I can run Cactus with the mesos batch system and the AWS job store:
```
cactus aws:us-west-2:adamnovak-jobstore ./examples/evolverMammals.txt em.hal --batchSystem mesos --skipPreprocessor --consCores 4 --clean=never --logDebug
```

And that runs through for me and succeeds. So then I clean up:
```
toil destroy-cluster -z us-west-2a adamnovak-test-cluster
toil clean aws:us-west-2:adamnovak-jobstore
```

@glennhickey Are you sure that the problem isn't something to do with where you are getting the values to *put* into `CACTUS_MAX_MEMORY`/`CACTUS_DEFAULT_MEMORY`? It looks like you are pulling out values for options that Toil defined, and the error message looks like maybe those environment variables *are* being propagated, but that they might contain the string `"None"`:
```
            return max(min(int(os.environ['CACTUS_MAX_MEMORY']), memory_bytes), int(os.environ['CACTUS_DEFAULT_MEMORY']))
        ValueError: invalid literal for int() with base 10: 'None'
```

Maybe a `~/.toil/default.yaml` file with outdated contents is cramming `None`s into options unexpectedly? Or maybe something changed about what Toil defaults those values to (which isn't reflected in the commit I tested).

**Comment by glennhickey** [comment]
@adamnovak you're right -- this error is on Cactus's end.  Sorry for the noise!  I get these environment variables from `options.maxMemory` and `options.defaultMemory`.  They get default values on single machine and slurm, but are None on aws/mesos.  As you said, the error message about `"None"` reveals this.
